### PR TITLE
Fix reverse translation of non-constant values of OpCompositeConstruct

### PR DESCRIPTION
**Pull request by vmaksimo**
This patch introduces a way to use runtime values for structure fields.

### CODE REVIEW

**Comment by vmaksimo** [comment]
This is a split from #2252, should be merged first

**Review by asudarsa — APPROVED** [review]
LGTM. Thanks

File: lib/SPIRV/SPIRVReader.cpp [hunk 1]
```diff
@@ -2234,9 +2234,14 @@ Value *SPIRVToLLVM::transValueWithoutDecoration(SPIRVValue *BV, Function *F,
     auto *CC = static_cast<SPIRVCompositeConstruct *>(BV);
     auto Constituents = transValue(CC->getOperands(), F, BB);
     std::vector<Constant *> CV;
+    bool HasRtValues = false;
     for (const auto &I : Constituents) {
-      CV.push_back(dyn_cast<Constant>(I));
+      auto *C = dyn_cast<Constant>(I);
+      CV.push_back(C);
+      if (!HasRtValues && C == nullptr)
+        HasRtValues = true;
```
Comment (asudarsa) [inline lib/SPIRV/SPIRVReader.cpp:2242]:
```suggestion
      if ((C = dyn_cast<Constant>(I)) != nullptr)
        CV.push_back(C);
      HasRtValues = true;
```

**Review by asudarsa — APPROVED** [review]
LGTM. Thanks

File: lib/SPIRV/SPIRVReader.cpp [hunk 2]
```diff
@@ -2246,7 +2251,22 @@ Value *SPIRVToLLVM::transValueWithoutDecoration(SPIRVValue *BV, Function *F,
...
       auto *ST = cast<StructType>(transType(CC->getType()));
-      return mapValue(BV, ConstantStruct::get(ST, CV));
+      if (!HasRtValues)
+        return mapValue(BV, ConstantStruct::get(ST, CV));
+
+      AllocaInst *Alloca = new AllocaInst(ST, SPIRAS_Private, "", BB);
+
+      // get pointer to the element of structure
+      // store the result of argument
+      for (size_t I = 0; I < CV.size(); I++) {
```
Comment (asudarsa) [inline lib/SPIRV/SPIRVReader.cpp:2261]:
```suggestion
      for (size_t I = 0; I < Constituents.size(); I++) {
```
for sake of clarity

**Review by asudarsa — APPROVED** [review]
Just one change requested. Thanks

**Pull request merged**